app/src/main/assets/chart.html

```diff
@@ -15,12 +11,13 @@
     <tr>
       {% for tile in tileRow %}
```

Comment:
I worry that this would be significantly slower with JavaScript, and would involve writing quite a bit more code than we have here.  Do you think so?  If you like, you could try experimenting with a simple example, and seeing how much JS you need to write to get it to work and how fast it is.  If it is at least as fast and can be expressed in code that isn't longer or more complex than the template, I'd support that change.

Again, however, I should caution that this isn't a priority right now — we don't need to redesign how this works for the field trip.  It could be a longer-term design question to return to after the trip, when we have time for some experimentation and further analysis.